libsql-client/src/lib.rs

```diff
@@ -232,19 +229,23 @@ impl Connection {
     /// * `stmts` - SQL statements
     pub async fn batch(
         &self,
-        stmts: impl IntoIterator<Item = impl Into<String>>,
-    ) -> Result<Vec<ResultSet>> {
+        stmts: impl IntoIterator<Item = impl Into<Statement>>,
+    ) -> Result<Vec<QueryResult>> {
         let mut headers = Headers::new();
         headers.append("Authorization", &self.auth).ok();
-        let stmts: Vec<String> = stmts
-            .into_iter()
-            .map(|s| format!("\"{}\"", s.into()))
-            .collect();
+        // FIXME: serialize and deserialize with existing routines from sqld
+        let mut body = "{\"statements\": [".to_string();
+        let mut stmts_count = 0;
```

Comment:
Ultra nit: It seems that boolean flag like `is_empty` would be enough and we don't really need to count the elements.

Reply:
We use `stms_count` later to validate the number of returned results, and `stmts` is an iterator, so there's no easy way of calling `is_empty` or `len` on it